// Params for initializing renderer.

package genart.core;

// Parameters limiting mutations.
message MutationParams {
  extensions 100 to 199;
  // 100 - PolyMutationParams
}

// Manually assigned points of interest.
message FocalPoint {
  // Normalized x coordinate.
  optional double x = 1;

  // Normalized y coordinate.
  optional double y = 2;

  // Normalized radius.
  optional double radius = 3;

  // Strength of the boost within the focal point.
  optional double strength = 4;

  // TODO(piotrf): add different types of boost?
}

message EffectParams {
  // Maximum generations to run algorithm.
  optional int32 max_generations = 1;

  // If set, run algorithm until fitness (from 0 to 1) drops below this
  // threshold.  A reasonable value is probably between 0.1 and 0.2.
  optional double fitness_threshold = 2;
  
  // Mutation rates.
  optional MutationParams mutation_params = 3;

  // Any focal points for evaluation, these are best put on faces.
  repeated FocalPoint focal_point = 4;
}